### ISSUE TITLE
Add authentication options, e.g. OpenID, IMAP, LDAP

### ISSUE THREAD

**Issue by imsoftware** [pagekit/pagekit#728]
As an end user, i want to login with my known username and password, so that i don't need to notice new data.

As an admin, i want to connect my existing user management system, so that i needn't to create new accounts. 

Feature request: May you add additional authentication option, e.g.:

- [ ] Auth via IMAP
- [ ] OpenID connect 
- [ ] LDAP

**Comment by MalteScharenberg** [comment]
I think this would be great as an extension.

**Comment by imsoftware** [comment]
Yes, maybe it's better to provide an extension than to put it in the core. 
In august i have no time to develop. 
Can somebody else start the extension?
I can join in september.

**Comment by brzuchal** [comment]
That would be great especially `LDAP` auth.

**Comment by nickdring** [comment]
+1
